microsoft/hyper-v/default.nix

```diff
@@ -0,0 +1,104 @@
+{ config, pkgs, options, lib, ... }:
+
+with lib; {
+
+  options = {
+    hyperv.enhancedSession = mkOption
+      {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enables enhanced hyperv session over RDP
+          Enable Guest Services on Hyper-V Management settings and
+          don't forget to enable access over VSock on a Admin powershell with:
+
+          % Set-Vm -VmName <VM> -EnhancedSessionTransportType HvSocket
+        '';
+      };
+
+    hyperv.video = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enables Display Manager support on HyperV
+      '';
+    };
+
+    hyperv.defaultWindowManager = mkOption
+      {
+        type = types.separatedString " ";
+        default = "${config.services.xserver.displayManager.sessionData.wrapper}";
```

Comment:
Maybe cc @jtojnar if you open the PR towards nixpkgs regarding this option.

Reply:
I didn't have much success in this setting tho - I had to create `~/startwm.sh` executable to make it work hehe